cmake/detect-arch.cmake

```diff
@@ -16,6 +16,8 @@ elseif(MSVC)
     elseif ("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "ARM64")
         set(ARCH "aarch64")
     endif()
+elseif(EMSCRIPTEN)
+    set(ARCH "wasm32")
```

Comment:
Maybe a comment similar to this should be added?
```cmake
// Detect Emscripten in CMake for vcpkg which does not set __EMSCRIPTEN__ preprocessor.
set(ARCH "wasm32")
```

Reply:
I don't think this comment is correct, see response on PR itself.